earthmover/tests/earthmover.yaml

```diff
@@ -262,6 +265,79 @@ transformations:
           entity: species
           count_column: total_count
 
+  {% set months = ["January","February","March","April","May","June","July","August","September","October","November","December"] %}
+  cities_avg_weather_wide:
+    source: $sources.cities
+    operations:
+      - operation: keep_columns
+        columns:
+          - city
+          - state
+          - weather
+      - operation: add_columns
+        columns:
+          {% for month in months %}
+          avg_days_rain_{{month}}: "{%raw%}{{weather.avg_days_of_rain.{%endraw%}{{month}}{%raw%}}}{%endraw%}"
+          avg_high_temp_{{month}}: "{%raw%}{{weather.temperatures.avg_highs.{%endraw%}{{month}}{%raw%}}}{%endraw%}"
+          avg_low_temp_{{month}}: "{%raw%}{{weather.temperatures.avg_lows.{%endraw%}{{month}}{%raw%}}}{%endraw%}"
+          {% endfor %}
+          avg_days_rain_year: "{%raw%}{{{%endraw%}{%for month in months%}weather.avg_days_of_rain.{{month}}|int{%if not loop.last%} + {%endif%}{%endfor%}{%raw%}}}{%endraw%}"
+          avg_high_temp: "{%raw%}{{(({%endraw%}{%for month in months%}weather.temperatures.avg_highs.{{month}}|int{%if not loop.last%} + {%endif%}{%endfor%}{%raw%})/12)|int}}{%endraw%}"
+          avg_low_temp: "{%raw%}{{(({%endraw%}{%for month in months%}weather.temperatures.avg_lows.{{month}}|int{%if not loop.last%} + {%endif%}{%endfor%}{%raw%})/12)|int}}{%endraw%}"
+      - operation: drop_columns
+        columns:
+          - weather
+  cities_area_codes:
+    source: $sources.cities
+    operations:
+      - operation: keep_columns
+        columns:
+          - city
+          - state
+          - area_codes
+      - operation: flatten
+        flatten_column: area_codes
+        value_column: area_code
+      - operation: group_by
+        group_by_columns:
+          - city
+          - state
+        create_columns:
+          num_area_codes: count()
+  cities_zip_codes:
+    source: $sources.cities
+    operations:
+      - operation: keep_columns
+        columns:
+          - city
+          - state
+          - zip_codes
+      - operation: flatten
+        flatten_column: zip_codes
+        value_column: zip_code
+      - operation: group_by
+        group_by_columns:
+          - city
+          - state
+        create_columns:
+          num_zip_codes: count()
+  cities_code_stats:
+    source: $transformations.cities_area_codes
+    operations:
+      - operation: join
+        sources:
+          - $transformations.cities_zip_codes
+        join_type: inner
+        left_keys:
+          - city
+          - state
+        right_keys:
+          - city
+          - state
+      - operation: debug
+        function: head
+        rows: 2
+        transpose: True
```

Comment:
Is there a mechanism somewhere that checks that this produces the expected output?

Reply:
Good point, currently no. But including this in the test suite at least checks that the operation doesn't error.

I'm not sure how we would capture the output to test it, maybe @jayckaiser has thoughts on that. I also think it could be left as a future improvement to the test suite.

Reply:
Right, the _data_ output (the JSONL files generated by earthmover in the `output/` directory) is compared against `expected/`. I think John was asking about the _CLI_ output that this particular `debug` operation _should_ produce (if it's functioning correctly).